services/apps/portal/src/application-review/application-review.service.ts

```diff
@@ -103,4 +124,57 @@ export class ApplicationReviewService {
 
     return await this.applicationReviewRepository.save(applicationReview);
   }
+
+  verifyComplete(
+    applicationReview: ApplicationReview,
+  ): CompletedApplicationReview {
+    if (
+      !applicationReview.localGovernmentFileNumber ||
+      !applicationReview.firstName ||
+      !applicationReview.lastName ||
+      !applicationReview.position ||
+      !applicationReview.department ||
+      !applicationReview.phoneNumber ||
+      !applicationReview.email
+    ) {
+      throw new BaseServiceException('Contact information not complete');
+    }
+
+    if (applicationReview.isOCPDesignation === null) {
+      throw new BaseServiceException('OCP information not complete');
+    }
+
+    if (applicationReview.isOCPDesignation) {
+      if (
+        !applicationReview.OCPBylawName ||
+        !applicationReview.OCPDesignation ||
+        applicationReview.OCPConsistent === null
+      ) {
+        throw new BaseServiceException('OCP information not complete');
+      }
+    }
+
+    if (applicationReview.isSubjectToZoning === null) {
+      throw new BaseServiceException('Zoning information not complete');
+    }
+
+    if (applicationReview.isSubjectToZoning) {
+      if (
+        !applicationReview.zoningBylawName ||
+        !applicationReview.zoningDesignation ||
+        !applicationReview.zoningMinimumLotSize ||
+        applicationReview.isZoningConsistent === null
+      ) {
+        throw new BaseServiceException('Zoning information not complete');
+      }
+    }
+
+    if (applicationReview.isAuthorized === null) {
+      throw new BaseServiceException('Review authorization needs to be set');
+    }
+
+    //TODO: Where do we verify the documents are uploaded?
```

Comment:
I thought that verification of documents would check that at least one document of each required type is attached. However, have no idea what is required other than certificate of title

Reply:
Added it here, I was on the edge of not passing the application to the application review service, but I think its easier to have all the validation checks done in a single place.